index.js

```diff
@@ -93,6 +93,27 @@ function checkTopicQueryParameter(req, res, next) {
     next();
 }
 
+
+function checkQoSParameter(req,res,next){
+
+  if(!req.body.qos){
+    req.body.qos=0
+  }
+  if( 0 > parseInt(req.body.qos, 10) || parseInt(req.body.qos, 10)  >2){
```

Comment:
No need to parse it twice. Please use variable:
let qos = parseInt(req.body.qos, 10)